doc/examples/reconst_dki.py

```diff
@@ -48,99 +52,99 @@
 
 """
-.. figure:: HARDI193_bvalues.png
-   :align: center
-   **b-values of the loaded dataset**.
-
-From the figure above, we can check that the loaded dataset contains three
-non-zero b-values as required for DKI. However the highest b-value of 3500
-$s.mm^{-2}$ is higher than normally used on DKI. Since DKI neglects diffusion
-signal components higher than the 4th order KT, a upper bound of b-value < 3000
-$s.mm^{-2}$ is normally implied to decrease the influence of higher order terms
-[Jensen2010]_. Following this, we discard the b-value shell of 3500 $s.mm^{-2}$
-before DKI fitting.
+Since the diffusion kurtosis models involves the estimation of a large number
+of parameters [TaxCMW2015]_ and since the non-Gaussian components of the
+diffusion signal are more sensitivity to artefacts [NetoHe2012]_, a fundamental
+data pre-processing step for diffusion kurtosis fitting is to denoise our data.
+For this, we use Dipy's non-local mean filter (see
+:ref:`example-denoise-nlmeans`). Note that, since the HCP-like has a large
+number of diffusion-weigthed volume, this procedure can take some minutes to
+compute.
 """
 
-select_ind = gtab.bvals < 3000
-selected_bvals = gtab.bvals[select_ind]
-selected_bvecs = gtab.bvecs[select_ind, :]
-selected_data = data[:, :, :, select_ind]
+sigma = estimate_sigma(data, N=4)
+den = nlmeans(data, sigma=sigma, mask=mask)
 
 """
-The selected b-values and gradient directions are then converted to Dipy's
-GradientTable format.
+To avoid future reprocessing of the non-local mean field, below we save the
+denoised version of the HCP-like data.
 """
 
-from dipy.core.gradients import gradient_table
+nib.save(nib.Nifti1Image(den, affine), 'denoised_cenir_multib.nii.gz')
 
-gtab = gradient_table(selected_bvals, selected_bvecs)
-
-"""
-Before fitting the data some data pre-processing is done. First, we mask and
-crop the data to avoid calculating Tensors on the background of the image.
+""" 
+Finally, we compute a final version of the brain mask and crop the denoised
+data to avoid unnecessary calculations on the background of the image.
 """
 
-from dipy.segment.mask import median_otsu
-
-maskdata, mask = median_otsu(selected_data, 3, 1, True,
-                             vol_idx=range(10, 50), dilate=2)
+maskdata, mask = median_otsu(den, 4, 2, True, vol_idx=[0, 1], dilate=1)
 
 """
-Now that we have prepared the datasets we can go forward with the voxel
-reconstruction. This can be done by first instantiate the DiffusinKurtosisModel
-in the following way.
+Now that we have loaded and prepared the datasets we can go forward with the
+voxel reconstruction. This can be done by first instantiating the
+DiffusionKurtosisModel in the following way:
 """
 
 dkimodel = dki.DiffusionKurtosisModel(gtab)
 
 """
-Fitting the data is very simple. We just need to call the fit method of the
-DiffusinKurtosisModel in the following way:
+To fitting the data using the defined ``dkimodel``, we just need to call the
```

Comment:
=> "To fit the data using the defined model object, we call the fit function of this object:"